core/templates/components/graph-services/graph-layout.service.spec.ts

```diff
@@ -0,0 +1,904 @@
+      }
+    ];
+    expect(sgls.getAugmentedLinks(nodeData, links2)).toEqual(augmentedLinks);
+  });
+
+  it('should return when x, y label of source and target are same', () => {
+    let nodeData = {
+      State1: {
+        depth: 0,
+        offset: 0,
+        reachable: true,
+        x0: 0.07250000000000001,
+        y0: 0.15333333333333335,
+        xLabel: 0.1625,
+        yLabel: 0.23333333333333334,
+        id: 'State1',
+        label: 'State1',
+        height: 0.16,
+        width: 0.18000000000000002,
+        reachableFromEnd: false
+      },
+      State2: {
+        depth: 1,
+        offset: 0,
+        reachable: true,
+        x0: 0.07250000000000001,
+        y0: 0.42000000000000004,
+        xLabel: 0.1625,
+        yLabel: 0.23333333333333334,
+        id: 'State2',
+        label: 'State2',
+        height: 0.16,
+        width: 0.18000000000000002,
+        reachableFromEnd: false
+      },
+      State3: {
+        depth: 1,
+        offset: 1,
+        reachable: true,
+        x0: 0.29750000000000004,
+        y0: 0.42000000000000004,
+        xLabel: 0.3875,
+        yLabel: 0.5,
+        id: 'State3',
+        label: 'State3',
+        height: 0.16,
+        width: 0.18000000000000002,
+        reachableFromEnd: false
+      },
+      State4: {
+        depth: 2,
+        offset: 0,
+        reachable: true,
+        x0: 0.07250000000000001,
+        y0: 0.6866666666666666,
+        xLabel: 0.1625,
+        yLabel: 0.7666666666666666,
+        id: 'State4',
+        label: 'State4',
+        height: 0.16,
+        width: 0.18000000000000002,
+        reachableFromEnd: false
+      }
+    };
+
+
+    expect(sgls.getAugmentedLinks(nodeData, links2)).toBe(undefined);
+  });
+
+  it('should get graph witdh and height', () => {
+    let graphWidth = sgls.getGraphWidth(4, 15);
+    let graphHeight = sgls.getGraphHeight(nodeData1);
+
+
+    expect(graphWidth).toBe(630);
+    expect(graphHeight).toBe(420);
+  });
+
+  it('should compute layout and get last computed arrangement', () => {
+    let nodes: GraphNodes = {
+      State1: 'State1',
+      State2: 'State2',
+      State3: 'State3',
+      State4: 'State4',
+      State5: 'State5',
+      State6: 'State6',
+      State7: 'State7',
+      State8: 'State8',
+      State9: 'State9',
+      Orphaned: 'Orphaned'
+    };
+
+    let initNodeId: string = 'State1';
+    let finalNodeIds: string[] = ['State4'];
+
+    expect(sgls.getLastComputedArrangement()).toEqual(null);
+    expect(sgls.computeLayout(nodes, links1, initNodeId, finalNodeIds)).toEqual(
+      nodeData1);
+    expect(sgls.getLastComputedArrangement()).toEqual(nodeData1);
+  });
+
+  it('should get graph boundaries', () => {
+    let graphBoundaries = {
+      bottom: 5.873333333333333,
+      left: -4.9275,
+      right: 6.04,
+      top: -4.873333333333333
+    };
+    expect(sgls.getGraphBoundaries(nodeData1)).toEqual(graphBoundaries);
+  });
+
+  it('should modify position values', () => {
```

Comment:
What are the requirements for modifyPositionValues? List them, and test according to those requirements.

Reply:
Changed the test